Display cleanup?

caslam opened the pull request:
Took some extra variables and stuff out of display files. Also I put the config file back because it was kind of helpful to have the screen defined before Display.h, but now you have to change the screen type there instead of Main.h.

caslam committed:
Removed enum and some extra variables
Change screen with #define instead

caslam committed:
Shortened thermistor printing method a bit

caslam committed:
Slapped config.h back in

caslam committed:
Updated docs

Chase-Arline commented:
This documentation system is really annoying and seems unnecessary. 
You only changed like 3-4 source files but the documentation makes you commit like 900+ total files?

Chase-Arline commented on Firmware/Main/config.h:
```diff
@@ -0,0 +1,28 @@
+
+/// Defines the type of screen to generate in Display.ino.
+/// \note NOTE: COMMENT THIS OUT TO CHANGE DISPLAY TYPE TO SPEEDOMETER!
+#define USE_DEBUGGING_SCREEN
```
good use of config.h and ifdef!

Chase-Arline commented:
Have you tested the code when you have `USE_DEBUGGING_SCREEN` defined as well as when you do not have it defined?

s-syasir commented:
> This documentation system is really annoying and seems unnecessary. You only changed like 3-4 source files but the documentation makes you commit like 900+ total files?

Good point, yeah the issue is that it updates all HTML files and modifies them, I guess if there was a way to just avoid those changes? Or perhaps, when committing, dont rerun doxygen, until significant changes are made and then just rerun and have a specific commit tagged as Doxygen

Chase-Arline commented:
What does doyxgen give us that normal commenting documentation doesn't? 
I am not sure the pros are worth the con of maintaining it.

s-syasir commented:
> What does doyxgen give us that normal commenting documentation doesn't? I am not sure the pros are worth the con of maintaining it.

Good point. The comments are there, I supposed people should just read those and make sense of it. The only advantage is that the HTML files are a bit more organized and easier to read, but yeah, the 900 file commits are annoying to deal with for sure

s-syasir commented:
So yeah, Cassandra, if you want, once we review this change and approve it, do another PR for just removing the Doxygen files.

caslam commented:
Oh ok! The Doxygen stuff was actually very painless to run, but yeah I'm not sure if people are actually using it.

Also yes the USE_DEBUGGING_SCREEN has been tested!

Chase-Arline commented:
did you check for merge conflicts already? 
git checkout main
git merge display
to check

caslam commented:
No conflicts :(

Chase-Arline commented:
Cool, no conflicts is good.

On Sun, Apr 2, 2023, 7:55 AM caslam ***@***.***> wrote:

> No conflicts :(
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Washington-Superbike/ARHMA2021/pull/6#issuecomment-1493363865>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANB6FNKILN2FBBKUK6MOSS3W7GHMRANCNFSM6AAAAAAWQEUGWM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

caslam committed:
Update for speedometer screen

Added the main battery voltage and made RPM a bit smaller to make room
for other info. Speed doesn't reprint until it changes.

caslam commented:
OK last thing, the battery voltage needs to be on there. Sorry for leaving this open forever

@MatthewDKLe could you please give this a check if you have time?

MatthewDKLe reviewed (approved):
Great work on the display cleanup Cassandra. I'm more than happy to merge this with the main branch.
